dispatches/models/fossil_case/properties/thermaloil_properties.py

```diff
@@ -0,0 +1,430 @@
+                                    "for state block is not zero during "
+                                    "initialization.")
+
+        opt = get_solver(solver, optarg)
+
+        # ---------------------------------------------------------------------
+        # Solve property correlation
+        results = solve_indexed_blocks(opt, [blk])
+        if results.solver.termination_condition \
+                == TerminationCondition.optimal:
+            init_log.info('{} Initialisation Step 1 Complete.'.
+                          format(blk.name))
+        else:
+            init_log.warning('{} Initialisation Step 1 Failed.'.
+                             format(blk.name))
+
+        init_log.info('Initialization Step 1 Complete.')
+
+        # ---------------------------------------------------------------------
+        if state_vars_fixed is False:
+            # release state vars fixed during initialization if control
+            # volume didn't fix the state vars
+            if hold_state is True:
+                return flags
+            else:
+                blk.release_state(flags)
+        init_log.info('Initialization Complete.')
+
+    def release_state(blk, flags, outlvl=idaeslog.NOTSET):
+        # Method to release states only if explicitly called
+
+        if flags is None:
+            return
+        # Unfix state variables
+        revert_state_vars(blk, flags)
+        init_log.info('State Released.')
+
+
+@declare_process_block_class("ThermalOilStateBlock",
+                             block_class=_StateBlock)
+class ThermalOilStateBlockData(StateBlockData):
+    """
+    Property package for Thermal oil properties
+    """
+
+    def build(self):
+        """
+        Callable method for Block construction
+        """
+        super(ThermalOilStateBlockData, self).build()
+
+        # Create state variables
+        self._make_state_vars()
+
+        # Create required properties
+        self._make_prop_vars()
+
+        # Create constraints
+        self._make_constraints()
+
+    def _make_state_vars(self):
+        """Declare the necessary state variable objects."""
+        self.flow_mass = Var(initialize=100.0,
+                             domain=Reals,
+                             doc="Total mass flow [Kg/s]",
+                             units=pyunits.kg/pyunits.s)
+        self.temperature = Var(initialize=523,
+                               domain=Reals,
+                               doc="Temperature of thermal oil [K]",
+                               bounds=(260, 616),
+                               units=pyunits.K)
+        self.pressure = Var(initialize=101325,
+                            domain=Reals,
+                            doc="Pressure [Pa]",
+                            units=pyunits.Pa)
+
+    def _make_prop_vars(self):
+        """Make additional variables for calculations."""
+
+        self.enthalpy_mass = Var(self.phase_list,
```

Comment:
Do we need to maintain consistency with IDAES nomenclature? If yes, I think this should `enth_mass`